File: app/src/main/java/net/radityalabs/contactapp/presentation/di/module/ActivityModule.java
package net.radityalabs.contactapp.presentation.di.module;

import android.app.Activity;

import net.radityalabs.contactapp.presentation.di.scope.ActivityScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by radityagumay on 4/11/17.
 */

@Module
public class ActivityModule {
    private Activity mActivity;

    public ActivityModule(Activity activity) {
        this.mActivity = activity;
    }

    @Provides
    @ActivityScope
    public Activity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/di/module/ActivityModule.kt
package net.radityalabs.contactapp.presentation.di.module

import android.app.Activity

import net.radityalabs.contactapp.presentation.di.scope.ActivityScope

import dagger.Module
import dagger.Provides

/**
 * Created by radityagumay on 4/11/17.
 */

@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    @ActivityScope
    fun provideActivity(): Activity {
        return mActivity
    }
}
